### PR TITLE
fix pkg/controller/deployment staticcheck

### PR DESCRIPTION
**Pull request by wawa0210**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Part of #92402

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/assign @soltysh

### CODE REVIEW

**Review by foxish — APPROVED** [review]
/lgtm

File: pkg/controller/deployment/sync.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func (dc *DeploymentController) getNewReplicaSet(d *apps.Deployment, rsList, old
 
 		if needsUpdate {
 			var err error
-			if d, err = dc.client.AppsV1().Deployments(d.Namespace).UpdateStatus(context.TODO(), d, metav1.UpdateOptions{}); err != nil {
+			if _, err = dc.client.AppsV1().Deployments(d.Namespace).UpdateStatus(context.TODO(), d, metav1.UpdateOptions{}); err != nil {
```
Comment (soltysh) [inline pkg/controller/deployment/sync.go:176]: This `d` is being used later on in code, so it looks to me this is wrong change.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/controller/deployment/sync.go (hunk 1 above)
Comment (wawa0210) [reply pkg/controller/deployment/sync.go:176]: From the code 

https://github.com/kubernetes/kubernetes/blob/da54185c9eb076a71962ddb5aded24811467dbb3/pkg/controller/deployment/sync.go#L149-L188

Here, the assignment of `d` will only be executed within the condition of `if existingNewRS != nil`, and then return directly to jump out of the function. The code segment used behind `d` is not within this logical scope, and the latter `d` is passed in by the current function

It seems like this is okay to modify

Comment (soltysh) [reply pkg/controller/deployment/sync.go:176]: Yeah, you're right, thx.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/retest

**Pull request merged**